presto-spi/src/main/java/io/prestosql/spi/connector/NamePart.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class NamePart
+{
+    private final String name;
+    private final boolean isDelimited;
+    private final String legacyName;
+    private final String caseNormalizedName;
+
+    public static NamePart createNamePart(String name)
+    {
+        if (name.startsWith("\"") && name.endsWith(("\""))) {
```

Comment:
But there is a chance that we can derive a `schema` from a `Session` right ? In that case do we need to use a `SqlParser` for the same (Won't SqlParser be a overkill for the same but it would avoid duplicating the logic)